NomadicNibbler/foodtruck-fe
# Feature/map view

## What does this PR do?

- The MapView component was created
   - A react-google-map/api library was installed which creates map components used to render the google map from google maps api
   - a .env file is used to hold the google map api key
   - The map is implemented with a center marker and cluster markers with truck icons to represent location
   - currently the locations being used are coming from a mockuser.js data file that is imported into App.js and set to state in componentDidMount (this will need to be changed later when we hook up the user form)
   - App was converted to a class component to hold state
   - Router was setup but currently has no functionality until userform is implemented and the truck list
- A Truck List button was added (no functionality yet)
- A change location button was added (no functionality yet)
- Basic styling was added

## How should it be tested?
-  pull the branch down and run npm start, visit localhost://3000/map to see the MapView component
- You should see a center marker and two truck icons
- change the view to check the responsive design
- in the mockuser data, if you change the lat/long, the markers on the map should change position

## What do the future iterations hold?

-  functionality will be added to the truck list button and change location buttons. 
  - truck list button will render a different component
  - change location will render the use form
- A radius feature will be added
- mockdata will be connected to the backend api

## Linked Issues
- #22 
- #28
